src/inmanta/server/services/orchestrationservice.py

```diff
@@ -814,9 +822,116 @@ async def _put_version(
             all_agents: set[str] = {res.agent for res in rid_to_resource.values()}
             all_agents.add(const.AGENT_SCHEDULER_ID)
 
+            type_to_agent: dict[str, str] = {str(res.resource_type): res.agent for res in rid_to_resource.values()}
+            LOGGER.debug(module_version_info)
+
             for agent in all_agents:
                 await self.agentmanager_service.ensure_agent_registered(env, agent, connection=connection)
 
+            async def register_code(
+                cm_version: int,
+                environment: uuid.UUID,
+                module_version_info: dict[str, "PythonModule"] | None,
+                type_to_module_data: dict[str, set[str]] | None,
+                type_to_agent: dict[str, str] | None,
+                connection: Connection,
+            ) -> None:
+                if not all([module_version_info, type_to_module_data, rid_to_resource, type_to_agent]):
+                    LOGGER.debug(
+                        "Cannot populate join table modules_for_agent. The following arguments are not set: %s"
```

Comment:
the join table means nothing to the end-user, I would formulate this in terms of the external api.